sigstore/sign.py

```diff
@@ -174,10 +177,9 @@ def _signing_cert(
 
     def sign(
         self,
-        input_: IO[bytes],
+        input_: IO[bytes] | Statement,
```

Comment:
Flagging for review: this puts a third-part type (in-toto-attestation's `Statement`) in our public API.

Normally I'd be neutral on this, but it has a few other "meh" factors:

1. `in-toto-attestation` is still pre-1.0, meaning that we probably shouldn't lash our stable semver'd public API to it;
2. `Statement` is a thin wrapper for a `protobuf` generated `protobuf.Message` class, which both leaks `protobuf` implementation details into the public API *and* may change soon-ish (see https://github.com/in-toto/attestation/issues/291).

Another option here would be to create our own `sigstore.intoto_or_whatever.Statement` type, which would be a thin constructable shim over `in_toto_attestation.Statement`.

Reply:
I'm not against a shim, but I don't see this causing a problem in practice. Even if/when they move to betterproto, the way users will construct a `Statement` will be pretty much identical. If the `in-toto-attestation` API completely changes before 1.0 to the point where it somehow becomes painful for our users (maybe their API takes different types?), we'll probably need to make breaking changes to the shim anyway.

Reply:
That's fair. I'll consider this a non-blocker for now, which means this PR is good to go.